packages/flutter_tools/test/commands/test_test.dart

```diff
@@ -47,18 +47,33 @@ void main() {
     testUsingContext('run a test when its name matches a regexp', () async {
       Cache.flutterRoot = '../..';
       final ProcessResult result = await _runFlutterTest('filtering', automatedTestsDirectory, flutterTestDirectory,
-        extraArgs: const <String>["--name", "inc.*de"]);
-      if (!result.stdout.contains("+1: All tests passed"))
-        fail("unexpected output from test:\n\n${result.stdout}\n-- end stdout --\n\n");        
+        extraArgs: const <String>['--name', 'inc.*de']);
+      if (!result.stdout.contains('+1: All tests passed'))
+        fail('unexpected output from test:\n\n${result.stdout}\n-- end stdout --\n\n');
       expect(result.exitCode, 0);
     });
 
     testUsingContext('run a test when its name contains a string', () async {
       Cache.flutterRoot = '../..';
       final ProcessResult result = await _runFlutterTest('filtering', automatedTestsDirectory, flutterTestDirectory,
-        extraArgs: const <String>["--plain-name", "include"]);
-      if (!result.stdout.contains("+1: All tests passed"))
-        fail("unexpected output from test:\n\n${result.stdout}\n-- end stdout --\n\n");        
+        extraArgs: const <String>['--plain-name', 'include']);
+      if (!result.stdout.contains('+1: All tests passed'))
+        fail('unexpected output from test:\n\n${result.stdout}\n-- end stdout --\n\n');
+      expect(result.exitCode, 0);
+    });
+
+    testUsingContext('test runs to completion', () async {
+      Cache.flutterRoot = '../..';
+      final ProcessResult result = await _runFlutterTest('trivial', automatedTestsDirectory, flutterTestDirectory,
+        extraArgs: const <String>['--verbose']);
+      if ((!result.stdout.contains('+1: All tests passed')) ||
```

Comment:
Yeah, I considered that, but the pain of having a regular expression seemed to outweigh the likelihood of catching a bug where these end up out of order somehow.